pyanaconda/modules/subscription/satellite.py

```diff
@@ -0,0 +1,161 @@
+
+    try:
+        # NOTE: we explicitely don't verify SSL certificates while
+        #       downloading the provisioning script as the Satellite
+        #       instance will have it's own self signed certs that
+        #       will only be trusted once the provisioning script runs
+        result = session.get(script_url, headers=headers,
+                             proxies=proxies, verify=False,
+                             timeout=constants.NETWORK_CONNECTION_TIMEOUT)
+        if result.ok:
+            provisioning_script = result.text
+            result.close()
+        else:
+            log.debug("subscription: server returned %i code when downloading"
+                      " Satellite provisioning script", result.status_code)
+            result.close()
+    except RequestException as e:
+        log.debug("subscription: can't download Satellite provisioning script"
+                  " from %s with proxy: %s. Error: %s", script_url, proxies, e)
+
+    if provisioning_script:
+        log.debug("subscription: writing Satellite provisioning script to file: %s",
+                  PROVISIONING_SCRIPT_DOWNLOAD_PATH)
+        with open(PROVISIONING_SCRIPT_DOWNLOAD_PATH, "wt") as f:
+            f.write(provisioning_script)
+        # successfully downloaded and writen out the Satellite provisioning script
+        return True
+    else:
+        # failed to download the Satellite provisioning script
+        return False
+
+
+def run_satellite_provisioning_script(sysroot):
+    """Run the Satellite provisioning script.
+
+    Each Satellite instance provides a provisioning script that will
+    enable the currently running environment to talk to the given
+    Satellite instance.
+
+    This mainly menas that the self-signed certificates of the given
+    Satellite instance will be installed to the system but also some
+    necessary changes to rhsm.conf.
+
+    Also as we need to provision both the installation *and* target system
+    to talk to Satellite we need to run the provisioning script twice,
+    once in the installation environment and once on the target system.
+
+    This is achieved by running this function repeatedly with the root
+    parameter set accordingly.
+
+    :param str sysroot: system root where to run the provisioning script
+    :return: True on success, False otherwise
+    :rtype: bool
+    """
+
+    # FIXME: do we need to copy the script to the chroot to execute ?
+    if os.path.exists(PROVISIONING_SCRIPT_DOWNLOAD_PATH):
+        log.debug("subscription: executing Satellite provisioning script.")
+        rc = util.execWithRedirect("bash", argv=[PROVISIONING_SCRIPT_DOWNLOAD_PATH], root=sysroot)
+        if rc == 0:
+            log.debug("subscription: satellite provisioning script executed successfully")
+            return True
+        else:
+            log.debug("subscription: satellite provisioning script executed with error")
+            return False
+        return True
```

Comment:
This is a dead code.

Reply:
Removed.

Reply:
The code hasn't changed.

Reply:
Should be fixed now.

Reply:
Still it is here... Starting to be a bit funny. It will stay here until end of Anaconda :D

Reply:
Should be gone now I guess ? ;-)